### DISCUSSION TITLE
How to use subsystem in MultiSystems class

### DISCUSSION THREAD

**Discussion by oskfuta** [deepmodeling/dpdata#576] [General]
I want to separate my dataset from Quantum espresso outputs into training set and validation set by using attribute `subsystem` 

```
import dpdata
import numpy as np

data = dpdata.MultiSystems.from_dir(
           dir_name="../P-*", file_name="scf.out", fmt="qe/pw/scf")

# number of total frames
n = 839

# number of validation 
t = 100

# random number of  validation_data
index_validation = np.random.choice(n,size=t,replace=False)       

# other indexes are training_data
index_training = list(set(range(n))-set(index_validation))
data_training = data.sub_system(index_training)
data_validation = data.sub_system(index_validation)

# all training data put into directory:"training_data" 
data_training.to_deepmd_npy('training_data')

# all validation data put into directory:"validation_data"
data_validation.to_deepmd_npy('validation_data')
```

However, it didn't work because there is an error about the unsupported attribute.

> AttributeError: 'MultiSystems' object has no attribute 'sub_system'

Is there other way to separate the dataset in such a case?

**Comment by njzjz** [comment]
To split the dataset, you can call [`train_test_split`](https://docs.deepmodeling.com/projects/dpdata/en/master/api/dpdata.html#dpdata.MultiSystems.train_test_split).

**Reply by oskfuta** [reply]
Thank you for your suggestion!
